pkg/controller/hibernation/hibernation_controller_test.go

```diff
@@ -289,6 +293,139 @@ func TestReconcile(t *testing.T) {
+		expectRequeueAfter time.Duration
+		expectedPowerState hivev1.ClusterPowerState
+	}{
+		{
+			name: "cluster due for hibernate no condition", // cluster that has never been hibernated and thus has no running condition
+			cd: cdBuilder.GenericOptions(testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "8h")).Build(
+				testcd.InstalledTimestamp(time.Now().Add(-10 * time.Hour))),
+			expectedPowerState: hivev1.HibernatingClusterPowerState,
+		},
+		{
+			name: "cluster not yet due for hibernate no condition", // cluster that has never been hibernated and thus has no running condition
+			cd: cdBuilder.GenericOptions(testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "12h")).Options(
+				testcd.InstalledTimestamp(time.Now().Add(-10 * time.Hour))).Build(),
+			expectRequeueAfter: 2 * time.Hour,
+			expectedPowerState: "",
+		},
+		{
+			name: "cluster with running condition due for hibernate",
+			cd: cdBuilder.GenericOptions(
+				testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "8h")).Build(
+				testcd.WithCondition(hibernatingCondition(corev1.ConditionFalse, hivev1.RunningHibernationReason, 9*time.Hour)),
+				testcd.InstalledTimestamp(time.Now().Add(-10*time.Hour))),
+			expectedPowerState: hivev1.HibernatingClusterPowerState,
+		},
+		{
+			name: "cluster with running condition not due for hibernate",
+			cd: cdBuilder.GenericOptions(
+				testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "20h")).Options(
+				testcd.WithCondition(hibernatingCondition(corev1.ConditionFalse, hivev1.RunningHibernationReason, 6*time.Hour)),
+			).Options(
+				testcd.InstalledTimestamp(time.Now().Add(-10 * time.Hour))).Build(),
+			expectRequeueAfter: 14 * time.Hour,
+			expectedPowerState: "",
+		},
+		{
+			name: "cluster waking from hibernate",
+			setupActuator: func(actuator *mock.MockHibernationActuator) {
+				actuator.EXPECT().MachinesRunning(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(false, nil)
+			},
+			cd: cdBuilder.GenericOptions(
+				testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "4h")).Options(
+				testcd.WithCondition(hibernatingCondition(corev1.ConditionTrue, hivev1.ResumingHibernationReason, 8*time.Hour)),
+			).Options(
+				testcd.InstalledTimestamp(time.Now().Add(-10*time.Hour)),
+				o.shouldRun).Build(),
+			expectedPowerState: hivev1.RunningClusterPowerState,
+			expectRequeueAfter: stateCheckInterval,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+			mockActuator := mock.NewMockHibernationActuator(ctrl)
+			mockActuator.EXPECT().CanHandle(gomock.Any()).AnyTimes().Return(true)
+			if test.setupActuator != nil {
+				test.setupActuator(mockActuator)
+			}
+			mockBuilder := remoteclientmock.NewMockBuilder(ctrl)
+			mockCSRHelper := mock.NewMockcsrHelper(ctrl)
+			actuators = []HibernationActuator{mockActuator}
```

Comment:
I know that this is carried over from the existing hibernate tests, and I don't expect you to do anything about it, but this makes it impossible to run the tests in parallel.